src/plugins/formulas/engine/register.js

```diff
@@ -0,0 +1,192 @@
+import staticRegister from '../../../utils/staticRegister';
+import { isUndefined } from '../../../helpers/mixed';
+import { warn } from '../../../helpers/console';
+import { PLUGIN_KEY } from '../formulas';
+import { DEFAULT_LICENSE_KEY, getEngineSettingsWithDefaultsAndOverrides } from './settings';
+
+/**
+ * Setups the engine instance. It either creates a new (possibly shared) engine instance, or attaches
+ * the plugin to an already-existing instance.
+ *
+ * @param {object} hotSettings Object containing the Handsontable settings.
+ * @param {string} hotId Handsontable guid.
+ * @returns {null|object} Returns the engine instance if everything worked right and `null` otherwise.
+ */
+export function setupEngine(hotSettings, hotId) {
+  const pluginSettings = hotSettings[PLUGIN_KEY];
+  const engineConfigItem = pluginSettings.engine;
+
+  if (isUndefined(engineConfigItem)) {
+    warn('Missing the required `engine` key in the Formulas settings. Please fill it with either an' +
```

Comment:
It's a just suggestion for the next iterations. It'd be nice to move the plugin's dependencies to the plugins static key. Like in the [Filters#L61](https://github.com/handsontable/handsontable/blob/master/src/plugins/filters/filters.js#L61.) for example.